src/widgets/modals/SearchForm.js

```diff
@@ -76,11 +91,40 @@ export const SearchFormComponent = ({
     [listConfig]
   );
 
+  const selectRecord = useMemo(() => {
+    if (isPatient) return patient => selectPatient(patient);
+    if (isPrescriber) return prescriber => selectPrescriber(prescriber);
+    return () => null;
+  }, [isPatient, isPrescriber]);
+
+  const lookupPatient = useCallback(params => {
+    setIsQuerying(true);
+    queryPatientApi(params).then(({ error: patientError, data: patientData }) => {
+      if (patientError) {
+        setIsError(true);
+        setError(patientError);
+      }
+      setData(patientData);
+      setIsQuerying(false);
```

Comment:
if you close the modal while querying is this component still mounted? I think it will either lead to memory leaks with trying to cause updates on unmounted components, or performance decreases and subtle bugs in the future if this component is just always mounted?

 I think you would need to create a new state or ref which would be set with a `useEffect` on mounting/unmounting which can be used to check if the component is still mounted before trying to update state. This gets pretty messy!

I think it's a lot easier to either use the redux store for state or alternatively use async/await rather than callbacks, although that's not a great solution

Reply:
Good point!

I don't think `await` will make a difference, but will try out a custom hook with `useEffect` as suggested!